src/routes/signup/signup.js

```diff
@@ -0,0 +1,38 @@
+const qs = require("querystring");
+const fs = require("fs");
+const path = require("path");
+
+const user = (request, response) => {
+    let body = '';
+    request.on('data', function (data) {
+        body = body + data;
+    })
+        .on('end', function () {
+            const post = qs.parse(body);
+            const result = {
+                status: "success",
+                post
+            };
+            const filename = JSON.parse(body).username;
+            const pathToUserFile = path.join(__dirname, "..", "..", "db", "users", `${filename}.json`);
+
+            fs.appendFile(
+                pathToUserFile,
+                body,
+                function (error) {
+                    if (error) {
+                        console.log(error);
+                    }
```

Comment:
looks like if file will not created you will return success response anyway

paste response here